Index: /Users/msekoranja/Work/workspace/hibernate-3.3.1.GA/core/src/main/java/org/hibernate/util/StringHelper.java
===================================================================
--- /Users/msekoranja/Work/workspace/hibernate-3.3.1.GA/core/src/main/java/org/hibernate/util/StringHelper.java	(revision 15825)
+++ /Users/msekoranja/Work/workspace/hibernate-3.3.1.GA/core/src/main/java/org/hibernate/util/StringHelper.java	(working copy)
@@ -384,6 +384,8 @@
 					return alias.substring( i );
 				}
 			}
+			// msekoran patch: no alphabetic characters in alias, prefix with some 
+			return "fix" + alias; 			
 		}
 		return alias;
 	}
