--- Python/lib/python2.7/threading.py        2017-01-06 01:19:50.000000000 +0000
+++ Python/lib/python2.7/threading.py 2017-01-06 01:23:59.000000000 +0000
@@ -165,7 +165,10 @@
         return true.
 
         """
-        me = _get_ident()
+        if _get_ident:
+            me = _get_ident()
+        else:
+            me = -1
         if self.__owner == me:
             self.__count = self.__count + 1
             if __debug__:
@@ -200,7 +203,9 @@
         There is no return value.
 
         """
-        if self.__owner != _get_ident():
+        if _get_ident and self.__owner != _get_ident():
+            raise RuntimeError("cannot release un-acquired lock")
+        elif not _get_ident and self.__owner != -1:
             raise RuntimeError("cannot release un-acquired lock")
         self.__count = count = self.__count - 1
         if not count:
