--- orbdeps.mak.orig	2010-08-09 09:33:25.000000000 +0000
+++ orbdeps.mak	2010-08-09 09:34:00.000000000 +0000
@@ -6,6 +6,38 @@
 IF_OSPLMODE_IS_DEV:=$(findstring dev, $(SPLICE_TARGET))
 IS_WINDOWS_DEBUG:=$(and $(IF_OSPLENV_IS_WIN32), $(IF_OSPLMODE_IS_DEV))
 
+ifeq ($(SPLICE_ORB), DDS_ACE_TAO_X_Y_Z)
+    ORB_MK_INCLUDE_NAME = tao15-OF
+    ORB_INCLUDE	  = "-I$(TAO_ROOT)" -I"$(ACE_ROOT)"
+    ORB_LDLIBS	      = "-L$(ACE_ROOT)/lib" -lACE -lTAO -lTAO_PortableServer -lTAO_AnyTypeCode
+    ORB_IDL_COMPILER = $(ACE_ROOT)/bin/tao_idl
+    ORB_COMPILER     = $(ACE_ROOT)/bin/tao_idl #only needed for compiling the corba-C++ testcases
+    ORB_IDL_FLAGS    = -Sp -Sd -si S.i -ci C.i -I"$(OSPL_HOME)/etc/idl"
+    ORB_IDL_OUTPUT   = -o
+    ORB_CPP_FLAGS    = -DACE_HAS_EXCEPTIONS
+    ORB_SELECT_FLAGS = -D$(SPLICE_ORB)
+ifneq (,$(findstring win32,$(SPLICE_TARGET)))
+    ORB_CXX_FLAGS    = -Wb,export_macro=$(DECL_PREFIX) -Wb,export_include=$(DECL_INCLUDE)
+endif
+
+# various rules to define ORB specific source and object files
+    ORB_TOP_OBJ      = $(TOPIC_IDL:%.idl=%C$(OBJ_POSTFIX)) $(TOPIC_IDL2:%.idl=%C$(OBJ_POSTFIX))
+    ORB_TOP_SRC      = $(TOPIC_IDL:%.idl=%C.cpp) $(TOPIC_IDL:%.idl=%S.cpp) $(TOPIC_IDL2:%.idl=%C.cpp) $(TOPIC_IDL2:%.idl=%S.cpp)
+    ORB_TOP_HDR      = $(ORB_TOP_SRC:%.cpp=%.h) $(ORB_TOP_SRC:%.cpp=%$(INLINESRC_POSTFIX)) $(ORB_TOP_SRC2:%.cpp=%.h) $(ORB_TOP_SRC2:%.cpp=%$(INLINESRC_POSTFIX))
+
+    ORB_API_OBJ      = $(DCPS_API_IDL:%.idl=%C$(OBJ_POSTFIX)) $(INT_IDL:%.idl=%C$(OBJ_POSTFIX))
+    ORB_API_SRC      = $(DCPS_API_IDL:%.idl=%C.cpp) $(DCPS_API_IDL:%.idl=%S.cpp) $(INT_IDL:%.idl=%C.cpp) $(INT_IDL:%.idl=%S.cpp)
+    ORB_API_HDR      = $(ORB_API_SRC:%.cpp=%.h) $(ORB_API_SRC:%.cpp=%$(INLINESRC_POSTFIX)) $(ORB_API_SRC2:%.cpp=%.h) $(ORB_API_SRC2:%.cpp=%$(INLINESRC_POSTFIX))
+
+    IDLPP_ORB_OBJ    = $(TOPIC_IDL:%.idl=%DcpsC$(OBJ_POSTFIX)) $(TOPIC_IDL2:%.idl=%DcpsC$(OBJ_POSTFIX))
+    IDLPP_ORB_SRC    = $(TOPIC_IDL:%.idl=%DcpsC.cpp) $(TOPIC_IDL:%.idl=%DcpsS.cpp) $(TOPIC_IDL2:%.idl=%DcpsC.cpp) $(TOPIC_IDL2:%.idl=%DcpsS.cpp)
+    IDLPP_ORB_HDR    = $(IDLPP_ORB_SRC:%.cpp=%.h) $(IDLPP_ORB_SRC:%.cpp=%$(INLINESRC_POSTFIX)) $(IDLPP_ORB_SRC2:%.cpp=%.h) $(IDLPP_ORB_SRC2:%.cpp=%$(INLINESRC_POSTFIX))
+
+    #TODO determine if correct
+    ORB_DLRL_API_OBJ = $(DLRL_API_IDL:%.idl=%C$(OBJ_POSTFIX))
+    ORB_DLRL_API_SRC = $(DLRL_API_IDL:%.idl=%C.cpp) $(DLRL_API_IDL:%.idl=%S.cpp)
+    ORB_DLRL_API_HDR = $(ORB_DLRL_API_SRC:%.cpp=%.h) $(ORB_DLRL_API_SRC:%.cpp=%.inl)
+endif
 ifeq ($(SPLICE_ORB), DDS_ACE_TAO_5_6_6)
     ORB_MK_INCLUDE_NAME = tao15-OF
     ORB_INCLUDE	     = "-I$(TAO_ROOT)" -I"$(ACE_ROOT)"
