--- AlarmSystemConfiguration.xml	2006-09-26 14:05:32.000000000 -0600
+++ AlarmSystemConfiguration.xml.acs	2006-09-26 14:07:36.000000000 -0600
@@ -4,5 +4,5 @@
                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
                     xsi:schemaLocation="urn:schemas-cosylab-com:Alarm:1.0 Alarm.xsd">
 
-	<configuration-property name="Implementation">CERN</configuration-property>
+	<configuration-property name="Implementation">ACS</configuration-property>
 </alarm-system-configuration>
