diff -Naur log4cpp-1.0-orig/include/log4cpp/Priority.hh log4cpp-1.0/include/log4cpp/Priority.hh
--- log4cpp-1.0-orig/include/log4cpp/Priority.hh	2005-10-17 02:03:16.000000000 -0600
+++ log4cpp-1.0/include/log4cpp/Priority.hh	2010-05-05 11:10:11.000000000 -0600
@@ -69,14 +69,19 @@
          **/
         typedef enum {EMERG  = 0, 
 		      FATAL  = 0,
+              EMERGENCY = 0,
                       ALERT  = 100,
                       CRIT   = 200,
+                      CRITICAL = 200,
                       ERROR  = 300, 
                       WARN   = 400,
+                      WARNING = 400,
                       NOTICE = 500,
                       INFO   = 600,
                       DEBUG  = 700,
-                      NOTSET = 800
+                      DELOUSE = 800,
+                      TRACE = 900,
+                      NOTSET = 1000
         } PriorityLevel;
 
         /**
diff -Naur log4cpp-1.0-orig/src/Priority.cpp log4cpp-1.0/src/Priority.cpp
--- log4cpp-1.0-orig/src/Priority.cpp	2005-10-17 02:03:16.000000000 -0600
+++ log4cpp-1.0/src/Priority.cpp	2010-05-05 11:08:10.000000000 -0600
@@ -14,28 +14,30 @@
 namespace log4cpp {
 
     namespace {
-        const std::string names[10] = {
-            "FATAL",
+        const std::string names[12] = {
+            "EMERGENCY",
 			"ALERT",
-			"CRIT",
+			"CRITICAL",
 			"ERROR",
-			"WARN",
+			"WARNING",
             "NOTICE",
 			"INFO",
 			"DEBUG",
+            "DELOUSE",
+            "TRACE",
 			"NOTSET",
 			"UNKNOWN" 
         };
     }    
 
-    const int log4cpp::Priority::MESSAGE_SIZE = 8;
+    const int log4cpp::Priority::MESSAGE_SIZE = 10;
     
 
     const std::string& Priority::getPriorityName(int priority) throw() {
          
         priority++;
         priority /= 100;
-        return names[((priority < 0) || (priority > 8)) ? 8 : priority];
+        return names[((priority < 0) || (priority > 10)) ? 10 : priority];
     }
 
     Priority::Value Priority::getPriorityValue(const std::string& priorityName) 
