--- AlarmSystemConfiguration.xml	2006-10-09 21:53:54.000000000 -0600
+++ AlarmSystemConfiguration.xml.acs	2006-10-09 21:57:07.000000000 -0600
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 <alarm-system-configuration xmlns="urn:schemas-cosylab-com:acsalarm-alarmservice:1.0" 
 		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"> 
-	<configuration-property name="Implementation">CERN</configuration-property>
+	<configuration-property name="Implementation">ACS</configuration-property>
 </alarm-system-configuration>
