diff -Naur TAO.org/ACE_wrappers/ace/os_include/sys/os_types.h TAO/ACE_wrappers/ace/os_include/sys/os_types.h
--- TAO.org/ACE_wrappers/ace/os_include/sys/os_types.h	2008-03-04 14:51:23.000000000 +0000
+++ TAO/ACE_wrappers/ace/os_include/sys/os_types.h	2009-08-25 14:15:39.000000000 +0000
@@ -89,7 +89,7 @@
    typedef offset_t ACE_LOFF_T;
 #elif defined (WIN32)
    typedef __int64  ACE_LOFF_T;
-#elif (defined (ACE_VXWORKS) && (ACE_VXWORKS <= 0x660)) || \
+#elif (defined (ACE_VXWORKS) && (ACE_VXWORKS <= 0x670)) || \
   defined (ACE_LYNXOS_MAJOR) || \
   (defined (ACE_OPENVMS) && !defined (_LARGEFILE)) || \
   defined (__TANDEM)
diff -Naur TAO.org/ACE_wrappers/ace/OS_NS_arpa_inet.cpp TAO/ACE_wrappers/ace/OS_NS_arpa_inet.cpp
--- TAO.org/ACE_wrappers/ace/OS_NS_arpa_inet.cpp	2008-03-04 14:51:23.000000000 +0000
+++ TAO/ACE_wrappers/ace/OS_NS_arpa_inet.cpp	2009-08-25 14:17:38.000000000 +0000
@@ -36,7 +36,7 @@
       addr->s_addr = ip_addr;  // Network byte ordered
       return 1;
     }
-#elif defined (ACE_VXWORKS) && (ACE_VXWORKS <= 0x660)
+#elif defined (ACE_VXWORKS) && (ACE_VXWORKS <= 0x670)
   // inet_aton() returns OK (0) on success and ERROR (-1) on failure.
   // Must reset errno first. Refer to WindRiver SPR# 34949, SPR# 36026
   ::errnoSet(0);
diff -Naur TAO.org/ACE_wrappers/ace/OS_NS_unistd.inl TAO/ACE_wrappers/ace/OS_NS_unistd.inl
--- TAO.org/ACE_wrappers/ace/OS_NS_unistd.inl	2008-05-14 18:15:31.000000000 +0000
+++ TAO/ACE_wrappers/ace/OS_NS_unistd.inl	2009-08-25 14:17:51.000000000 +0000
@@ -21,7 +21,7 @@
 
 #if defined (ACE_VXWORKS) || defined (ACE_HAS_WINCE)
 #  include "ace/os_include/os_unistd.h"
-#  if defined (ACE_VXWORKS) && (((ACE_VXWORKS >= 0x620) && (ACE_VXWORKS <= 0x660)) || defined (ACE_HAS_VXWORKS551_MEDUSA))
+#  if defined (ACE_VXWORKS) && (((ACE_VXWORKS >= 0x620) && (ACE_VXWORKS <= 0x670)) || defined (ACE_HAS_VXWORKS551_MEDUSA))
 #    if defined (__RTP__)
 #      include "ace/os_include/os_strings.h"
 #    else
