--- TAO.org/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/AVStreams_i.cpp	2006-04-24 15:53:27.000000000 +0200
+++ TAO/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/AVStreams_i.cpp	2007-02-15 10:16:17.000000000 +0100
@@ -117,6 +117,14 @@
             }
         }
     }
+  ACE_CATCH(AVStreams::noSuchFlow, ex)
+      {
+      ACE_RE_THROW; //ACS mod 2006-02
+      }
+  ACE_CATCH(CORBA::SystemException, ex)
+      {
+      ACE_RE_THROW; //ACS mod 2006-03
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_Basic_StreamCtrl::stop");
@@ -167,6 +175,16 @@
             }
         }
     }
+  ACE_CATCH(AVStreams::noSuchFlow,ex)
+      {
+      ACE_PRINT_EXCEPTION(ex, "TAO_Basic_StreamCtrl::start");
+      ACE_RE_THROW;    //ACS mod 2007-02
+      }
+  ACE_CATCH(CORBA::SystemException,ex)
+      {
+      ACE_PRINT_EXCEPTION(ex, "TAO_Basic_StreamCtrl::start");
+      ACE_RE_THROW;    //ACS mod 2007-02
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_Basic_StreamCtrl::start");
@@ -510,6 +528,14 @@
           ACE_TRY_CHECK;
         }
     }
+ ACE_CATCH(AVStreams::noSuchFlow, ex)
+      {
+      ACE_RE_THROW; //ACS mod 2007-02
+      }
+  ACE_CATCH(CORBA::SystemException, ex)
+      {
+      ACE_RE_THROW; //ACS mod 2006-03
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_Basic_StreamCtrl::stop");
@@ -548,6 +574,16 @@
           ACE_TRY_CHECK;
         }
     }
+  ACE_CATCH(AVStreams::noSuchFlow,ex)
+      {
+      ACE_PRINT_EXCEPTION(ex, "TAO_StreamCtrl::start");
+      ACE_RE_THROW;  //ACS mod 2007-02
+      }
+  ACE_CATCH(CORBA::SystemException,ex)
+      {
+      ACE_PRINT_EXCEPTION(ex, "TAO_StreamCtrl::start");
+      ACE_RE_THROW;    //ACS mod 2007-02
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_StreamCtrl::start");
@@ -3932,6 +3968,10 @@
           ACE_TRY_CHECK;
         }
     }
+  ACE_CATCH(CORBA::SystemException, ex)
+      {
+      ACE_RE_THROW; //ACS mod 2006-03
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_FlowConnection::stop");
@@ -3967,6 +4007,11 @@
           ACE_TRY_CHECK;
         }
     }
+  ACE_CATCH(CORBA::SystemException,ex)
+      {
+      ACE_PRINT_EXCEPTION(ex, "TAO_FlowConnection::start");   
+      ACE_RE_THROW;    //ACS mod 2007-02
+      }
   ACE_CATCHANY
     {
       ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION, "TAO_FlowConnection::start");
--- TAO.org/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/TCP.cpp	2006-05-16 16:06:38.000000000 +0200
+++ TAO/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/TCP.cpp	2006-12-14 17:01:31.000000000 +0100
@@ -656,6 +656,19 @@
                       -1);
 #endif /* TCP_NODELAY */
 
+//begin: BJE: (allign the sizes of the buffers)
+  int bufSize=BUFSIZ;
+  int s= sizeof (bufSize);
+  if(  this->peer ().get_option (SOL_SOCKET, /*IPPROTO_TCP,*/
+				  SO_RCVBUF,
+                                (void *) &bufSize,&s
+			     ) == -1)
+       bufSize=BUFSIZ;
+
+    
+    ((TAO_AV_TCP_Object*)(this->protocol_object_))->frame_.size (bufSize);
+//end: BJE
+
   // Called by the <Strategy_Acceptor> when the handler is completely
   // connected.
   ACE_INET_Addr addr;
--- TAO.org/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/TCP.h	2006-03-14 07:14:24.000000000 +0100
+++ TAO/ACE_wrappers/TAO/orbsvcs/orbsvcs/AV/TCP.h	2006-12-14 17:01:31.000000000 +0100
@@ -248,7 +248,7 @@
   /// end the stream.
   virtual int destroy (void);
 
-private:
+public:
   /// Pre-allocated memory to receive the data...
   ACE_Message_Block frame_;
 };
