diff -Naur TAO.org/ACE_wrappers/TAO/tao/Transport_Selection_Guard.cpp TAO/ACE_wrappers/TAO/tao/Transport_Selection_Guard.cpp
--- TAO.org/ACE_wrappers/TAO/tao/Transport_Selection_Guard.cpp	2007-03-01 15:01:51.000000000 +0100
+++ TAO/ACE_wrappers/TAO/tao/Transport_Selection_Guard.cpp	2007-08-14 11:55:17.000000000 +0200
@@ -4,7 +4,7 @@
 #include "tao/TSS_Resources.h"
 
 ACE_RCSID (tao,
-           Transport_Selection_Guard.cpp,
+           Transport_Selection_Guard,
            "$Id: Transport_Selection_Guard.cpp 77480 2007-03-01 14:01:51Z johnnyw $")
 
 TAO_BEGIN_VERSIONED_NAMESPACE_DECL
